File: app/src/main/java/com/zack/enderplan/injector/module/GuidePresenterModule.java
package com.zack.enderplan.injector.module;

import com.zack.enderplan.view.contract.GuideViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class GuidePresenterModule {

    private final GuideViewContract mGuideViewContract;

    public GuidePresenterModule(GuideViewContract guideViewContract) {
        mGuideViewContract = guideViewContract;
    }

    @Provides
    GuideViewContract provideGuideViewContract() {
        return mGuideViewContract;
    }
}

File: app/src/main/java/net/zackzhang/app/end/injector/module/GuidePresenterModule.kt
package net.zackzhang.app.end.injector.module

import dagger.Module
import dagger.Provides
import net.zackzhang.app.end.view.contract.GuideViewContract

@Module
class GuidePresenterModule(private val mGuideViewContract: GuideViewContract) {

    @Provides
    fun provideGuideViewContract() = mGuideViewContract
}
